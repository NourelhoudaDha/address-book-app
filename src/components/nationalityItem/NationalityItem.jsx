import { useCallback } from 'react'
import PropTypes from 'prop-types'

const NationalityItem = ({ onClick, value }) => {
  const memoizedUpdateNationality = useCallback(() => {
    onClick(value)
  }, [])
  return (
    <button
      onClick={memoizedUpdateNationality}
      className='hover:bg-light-blue-500 hover:border-transparent hover:shadow-lg group block rounded-lg p-4 border border-gray-200 flex items-center justify-center'
    >
      <div>{value}</div>
    </button>
  )
}

NationalityItem.propTypes = {
  onClick: PropTypes.func,
  value: PropTypes.string,
}

export default NationalityItem
