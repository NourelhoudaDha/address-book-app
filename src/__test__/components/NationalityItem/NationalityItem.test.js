import { render, cleanup } from '@testing-library/react'
import NationalityItem from '../../../components/nationalityItem/NationalityItem.jsx'
const mockOnClick = jest.fn()
const mockNationalityValue = 'ES'
afterEach(cleanup)
describe('NationalityItem', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <NationalityItem onClick={mockOnClick} value={mockNationalityValue} />
    )
    expect(
      asFragment(
        <NationalityItem onClick={mockOnClick} value={mockNationalityValue} />
      )
    ).toMatchSnapshot()
  })
})
