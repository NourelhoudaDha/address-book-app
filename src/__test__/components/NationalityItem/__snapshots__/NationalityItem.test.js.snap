// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NationalityItem should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="hover:bg-light-blue-500 hover:border-transparent hover:shadow-lg group block rounded-lg p-4 border border-gray-200 flex items-center justify-center"
  >
    <div>
      ES
    </div>
  </button>
</DocumentFragment>
`;
